yoyodyne/models/base.py

```diff
@@ -11,16 +11,19 @@
 from torch import nn, optim
 
 from .. import batches, defaults, evaluators, schedulers
+from .modules import base, check_encoder_compatibility
 
 
 class BaseEncoderDecoder(pl.LightningModule):
+    #  TODO: clean up type checking here.
```

Comment:
This seems like an important TODO but maybe you could say what's unclean about it?

Reply:
Oh yeah, this is a question. I'm curious about the typing rules for noting optionals. A lot of these variables functional as optional variables, but it doesn't seem like they're treated that way. Wanted to know the formal typing rules here.